src/Auth/Jwt.ts

```diff
@@ -119,20 +115,40 @@ export class Jwt implements IAccessToken {
 	 * @param {IJwtBody} body
 	 * @param {string} signature
 	 */
-	constructor (
-		public readonly header: IJwtHeader,
-		public readonly body  : IJwtBody,
-		public readonly signature?: string
-	) {
-		const withoutSignature = this.headerBase64() + '.' + this.bodyBase64();
-
-		this.unsignedData = withoutSignature;
-
-		if (this.signature == null) {
-			this.stringRepresentation = withoutSignature;
+	constructor (header: IJwtHeader, body  : IJwtBody, signature?: string);
+	constructor (header: IJwtHeader | string, body?: IJwtBody, signature?: string) {
+		if (typeof header === 'string') {
+			const stringRepresentation = header;
+			const parts = stringRepresentation.split('.');
+
+			if (parts.length !== 3) throw new Error('Wrong JWT format');
```

Comment:
I think it's better to throw `TypeError` here
```suggestion
			if (parts.length !== 3) throw new TypeError('Wrong JWT format');
```